napari/plugins/_builtins.py

```diff
@@ -32,4 +28,4 @@ def napari_get_reader(path: Union[str, List[str]]) -> ReaderFunction:
     callable
         function that returns layer_data to be handed to viewer._add_layer_data
     """
-    return _interal_reader_plugin
+    return array_return_to_layerdata_return(magic_imread)
```

Comment:
Just making sure I'm understanding: technically, shouldn't this return `None` if the file is not something that `magic_imread` can read? e.g. a `.csv` file? But you are leaving it like this because we might as well try it and have it crash rather than create a super complex checker?

Reply:
Good question :) 
Yes: because our own `napari_get_reader` plugin is last in line, we let it be a little special and try to read anything.  This is partially because `skimage.io` is somewhat flexible (e.g. even URLs work) and I didn't want to accidentally exclude something it _could_ read.
Note too: because we _also_ use a try/catch in [this line](https://github.com/napari/napari/blob/9ba11095e8a90f6056194249d539df9eb10c612c/napari/plugins/io.py#L47-L49) of `read_data_with_plugins`, if `magic_imread` throws an exception when reading the file, it will fail gracefully: catching and storing the exception, and still showing the "no plugin reader found" error.  (Users could also report that to us in #1024)